Spring-Boot/Spring-Boot-Academy/src/test/java/tw/waterball/judgegirl/springboot/academy/controllers/HomeworkControllerTest.java

```diff
@@ -159,20 +190,93 @@ public void testGetHomeworkProgress() throws Exception {
         homeworkProgressShouldIncludeTwoBestRecords(homework, homeworkProgress, bestRecord1, bestRecord2);
     }
 
+    @DisplayName("Given a homework consists of two problems [p1, p2] and two students [A, B]," +
+            "And student A achieved AC in p1 and CE in p2," +
+            "And student B achieved AC in p1 and didn't answer p2," +
+            "When getting students homework progress," +
+            "Then student A should get 100 scores in p1 and 0 scores in p2," +
+            "And student B should get 100 scores in p1 and 0 scores in p2,")
+    @Test
+    public void testGetStudentsHomeworkProgress() throws Exception {
+        var homework = createHomeworkWithProblems(PROBLEM1_ID, PROBLEM2_ID);
+        achieveBestRecord(PROBLEM1_ID, submission("AC").AC(1, 1, 100).build(studentA.id, PROBLEM1_ID, LANG_ENV));
+        achieveBestRecord(PROBLEM2_ID, submission("CE").CE(100).build(studentA.id, PROBLEM2_ID, LANG_ENV));
+        achieveBestRecord(PROBLEM1_ID, submission("AC").AC(1, 1, 100).build(studentB.id, PROBLEM1_ID, LANG_ENV));
+
+        var studentsHomeworkProgresses = getStudentsHomeworkProgress(homework.id, studentA.email, studentB.email).getProgress();
+
+        assertTrue(studentsHomeworkProgresses.containsKey(studentA.email));
+        assertTrue(studentsHomeworkProgresses.containsKey(studentB.email));
+        var progressA = studentsHomeworkProgresses.get(studentA.email);
+        var progressB = studentsHomeworkProgresses.get(studentB.email);
+        progressShouldHaveGrade(progressA, studentA, homework.problemIds);
+        progressShouldHaveGrade(progressB, studentB, homework.problemIds);
+    }
+
+    @DisplayName("Given a homework consists of two problems [p1, p2] and two students [A, B, C]," +
```

Comment:
1. ... `and three students [A, B, C]`
2. you miss the context "group students to the group"